src/main/scala/server/ServerApp.scala

```diff
@@ -8,40 +8,61 @@ import org.http4s.dsl.io._
 import org.http4s.implicits._
 import io.circe.generic.auto._
 import org.http4s.circe._
-import Domain.{Subscription, SubscriptionRequest}
+import Domain.{Subscription, SubscriptionRequest, PutRequest}
 
 import java.util.UUID
 import scala.concurrent.ExecutionContext.global
 
 object ServerApp extends IOApp {
 
-  implicit val decoder: EntityDecoder[IO, SubscriptionRequest] = jsonOf[IO, SubscriptionRequest]
+  implicit val decoderPost: EntityDecoder[IO, SubscriptionRequest] = jsonOf[IO, SubscriptionRequest]
+  implicit val decoderPut: EntityDecoder[IO, PutRequest] = jsonOf[IO, PutRequest]
 
   override def run(args: List[String]): IO[ExitCode] = {
-    var storage: Map[String, Subscription] = Map.empty
+    implicit var storage: Map[String, Subscription] = Map.empty
+
+    def existsId(id: String, request: Either[String, PutRequest])(implicit storage: Map[String, Subscription]): Either[String, PutRequest] = {
+      if (!storage.contains(id))
+        Left(s"The subscription with id $id does not exist + ${request.left.toOption.get}")
```

Comment:
There are two problems with `request.left.toOption.get`.

1. `request.left.toOption` will give `None` in case `request` is `Right`
2. `toOption.get` is dangerous for the case when `Option` is `None`, because it throws exception as we're becoming unsafe here. 

I'm no really sure what you want to do here. Maybe if you explain it, we could find a better solution.

Reply:
yeah I know, this is not very thought through... I just wanted all error messages concatenated, but that's maybe a different concern; I'll just remove it